scripts/gpu_support/nvidia/install_cuda_and_libraries.sh

```diff
@@ -0,0 +1,238 @@
+    echo "                                   cuDNN, see the EULA at"
+    echo "                                   https://docs.nvidia.com/deeplearning/cudnn/latest/reference/eula.html"
+    echo "  -t, --temp-dir /path/to/tmpdir   Specify a location to use for temporary"
+    echo "                                   storage during the installation of CUDA"
+    echo "                                   and/or other libraries (must have"
+    echo "                                   several GB available; depends on the number of installations)"
+}
+
+# Initialize variables
+cuda_eula_accepted=0
+cudnn_eula_accepted=0
+EASYSTACK_FILE=
+TEMP_DIR=
+
+# Parse command-line options
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        --help)
+            show_help
+            exit 0
+            ;;
+        --accept-cuda-eula)
+            cuda_eula_accepted=1
+            shift 1
+            ;;
+        --accept-cudnn-eula)
+            cudnn_eula_accepted=1
+            shift 1
+            ;;
+        -t|--temp-dir)
+            if [ -n "$2" ]; then
+                TEMP_DIR="$2"
+                shift 2
+            else
+                echo "Error: Argument required for $1"
+                show_help
+                exit 1
+            fi
+            ;;
+        *)
+            show_help
+            fatal_error "Error: Unknown option: $1"
+            ;;
+    esac
+done
+
+# Make sure EESSI is initialised
+check_eessi_initialised
+
+# Make sure that `EESSI-extend` will install in the site installation path EESSI_SITE_SOFTWARE_PATH
+export EESSI_SITE_INSTALL=1
+echo "EESSI_SITE_SOFTWARE_PATH=${EESSI_SITE_SOFTWARE_PATH}"
+
+# we need a directory we can use for temporary storage
+if [[ -z "${TEMP_DIR}" ]]; then
+    tmpdir=$(mktemp -d)
+else
+    mkdir -p ${TEMP_DIR}
+    tmpdir=$(mktemp -d --tmpdir=${TEMP_DIR} cuda_n_co.XXX)
+    if [[ ! -d "$tmpdir" ]] ; then
+        fatal_error "Could not create directory ${tmpdir}"
+    fi
+fi
+echo "Created temporary directory '${tmpdir}'"
+
+# use EESSI_SITE_SOFTWARE_PATH/.modules/all as MODULEPATH
+SAVE_MODULEPATH=${MODULEPATH}
+
+for EASYSTACK_FILE in ${TOPDIR}/easystacks/eessi-*CUDA*.yml; do
+    echo -e "Processing easystack file ${easystack_file}...\n\n"
+
+    # We don't want hooks used in this install, we need vanilla installations
+    touch "${tmpdir}"/none.py
+    export EASYBUILD_HOOKS="${tmpdir}/none.py"
+
+    # determine version of EasyBuild module to load based on EasyBuild version included in name of easystack file
+    eb_version=$(echo ${EASYSTACK_FILE} | sed 's/.*eb-\([0-9.]*\).*/\1/g')
+
+    # Load EasyBuild version for this easystack file _before_ loading EESSI-extend
+    module avail EasyBuild
+    module load EasyBuild/${eb_version}
+    module load EESSI-extend/${EESSI_VERSION}-easybuild
+
+    # Install modules in hidden .modules dir to keep track of what was installed before
+    MODULEPATH=${EESSI_SITE_SOFTWARE_PATH}/.modules/all
```

Comment:
Doesn't this cause `EESSI-extend` to be unloaded? I guess that only happens the next time you use use Lmod (and luckily that is happening in EB subshells)
```suggestion
    # (this action is temporary, and we do not call Lmod again within the current shell context, but in EasyBuild
    # subshells, so loaded modules are not automatically unloaded)
    MODULEPATH=${EESSI_SITE_SOFTWARE_PATH}/.modules/all
```

Reply:
Yeah, Lmod won't be happy after a hard reset of `$MODULEPATH`, but I don't think it matters here


If we want to avoid that, we can use the following below:

```shell
MODULEPATH=${EESSI_SITE_SOFTWARE_PATH}/.modules/all eb --dry-run-short ...